src/menus/hotkeyMenu.cpp

```diff
@@ -204,14 +205,15 @@ void HotkeyMenu::saveHotkeys()
     }
 
     // Read in all TextEntry values and update hotkeys
+    std::regex buttonIdExpression ("(\\[J([0-7])\\])?\\[B([0-9]|[12][0-9]|3[0-1])\\]"); // matches [J0] - [J7] (optional) and  [B0] - [B31]
```

Comment:
Raw string literals can most likely help making this a bit more readable: <https://www.geeksforgeeks.org/raw-string-literal-c/>

And I think  just testing for `[JB][0-9]+` might be good enough and simpler to read?

Reply:
Interesting. Did not know about raw string literals, although in that case I am not sure if getting rid of two blackslashes helps that much. Anyways, you might be right about simplifying the pattern, should be be good enough for now. So I went with that option. Wether it is really more readable now or not, might be a matter of taste though ;-)